addons/web/static/src/js/views/list/list_controller.js

```diff
@@ -607,6 +697,7 @@ var ListController = BasicController.extend({
             }));
             this.$selectionBox.appendTo(this.$buttons);
         }
+        this._renderHeaderButtons();
```

Comment:
FIXME, find a better place to call this,
HOWEVER: those buttons are causally related to selection change !

Reply:
@aab-odoo 
I have invistegated why updateSelectionBox is called in updateButtons
It's because the list can receive at update() a parameter to to keep the selection or not.
It makes the selection sensitive to any update, hence the total causality between updating the selection box and the header buttons

Reply:
seems acceptable :-)

Reply:
maybe add a sentence in the docstring